### ISSUE TITLE
Enhancement: [no-misused-spread] Option to allow string literals

### ISSUE THREAD

**Issue by syimyuzya** [typescript-eslint/typescript-eslint#10954]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-misused-spread/

### Description

The `allow` option of this rule can be configured to allow spreading strings with `{ from: 'lib', name: 'string' }`, but this does not cover string literals, and there is no way to specify things like "any string literal" using this option.

I propose that we add an option like `allowStringLiterals` in addition to `allow` (or make it possible to specify "any string literal" with `TypeOrValueSpecifier`), which allows spreading of just string literals.

### Fail

```TypeScript
let myStr: string = "String literals in this project is known to not contain grapheme clusters.";

[...myStr]; // not a string literal type, thus not covered by this option
```

### Pass

```TypeScript
[..."String literals in this project is known to not contain grapheme clusters."];

const myStr = "a string const";
[...myStr];
```

**Comment by bradzacher** [comment]
I'm not quite sure I understand - why should string literal typed values be exempted and special case? You did not explain the motivation for your request.

Why should such an option exist?

**Comment by syimyuzya** [comment]
> I'm not quite sure I understand - why should string literal typed values be exempted and special case? You did not explain the motivation for your request.
> 
> Why should such an option exist?

Yeah I've also realized that my proposal might not be the ideal way to solve the problem.

The real problem is: There has to be a way to configure this rule to (dis)allow *any string (including literals)*. (even if not in the way I proposed)

Here are the cases listed in the rule's documentation:

> 1. Spreading a Promise into an object. You probably meant to await it.
> 2. Spreading a function without properties into an object. You probably meant to call it.
> 3. Spreading an iterable (Array, Map, etc.) into an object. Iterable objects usually do not have meaningful enumerable properties and you probably meant to spread it into an array instead.
> 4. Spreading a string into an array. String enumeration behaviors in JavaScript around encoded characters are often surprising.
> 5. Spreading a class into an object. This copies all static own properties of the class, but none of the inheritance chain.
> 6. Spreading a class instance into an object. This does not faithfully copy the instance because only its own properties are copied, but the inheritance chain is lost, including all its methods.

Say, if I want to turn on/off this rule just for case 4 (spreading strings), I would naturally expect that this rule can be configured such, which isn't possible now. (same for the other 5 cases too)

**Comment by JoshuaKGoldberg** [comment]
> Yeah I've also realized that my proposal might not be the ideal way to solve the problem.

Since it's been a little over a month and nobody's posted a real concrete use case, I'm going ahead and closing it out. But if someone has an explanation for why they'd want it, please do go ahead and post here (if it's not yet auto-locked) or file a new issue. Cheers!

**Comment by Josh-Cena** [comment]
@JoshuaKGoldberg Sorry I didn't reply to this earlier. I believe the inability to allow `string` is a mistake. When we first designed this, we had separate options for `allowString`, `allowIterable`, etc., and I pointed out that we should have a generic `allow` option to accomodate all of these. I wasn't expecting `TypeOfValueSpecifier` to be unable to specify keyword types.

**Comment by JoshuaKGoldberg** [comment]
> I wasn't expecting `TypeOfValueSpecifier` to be unable to specify keyword types.

You can:

```json
{
  "rules": {
    "@typescript-eslint/no-misused-spread": ["error", {
      "allow": [
        {
          "from": "lib",
          "name": "string"
        }
      ]
    }]
  }
}
```

[The playground shows no complaints for string config](https://typescript-eslint.io/play/#ts=5.9.2&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3gVzSGAclALYgBc8mMAligOYDcAUMulmABaxRgb5rwAvPADaAOgm58RUgF0GQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFtLlZkiACa1kxaIgCGQ9GADakRNGgdokADSZw2HJPjwOAd1lztOiFnM7IAMxXdZkRgCMNZq1CaTuiR8nxUTADmkO46AL5hYAC67uGxEJHhQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false):

```ts
declare const userName: string;
const characters = [...userName];
```

Is this a docs issue, perhaps?

**Comment by Josh-Cena** [comment]
Oh! TIL. I thought `LibSpecifier` can only match things "in the lib". Perhaps this should be documented in https://typescript-eslint.io/packages/type-utils/type-or-value-specifier and/or have an example of allowing strings since often using `Intl.Segmenter` is overkill.
